SimMuon/GEMDigitizer/src/ME0PreRecoGaussianModel.cc

```diff
@@ -11,41 +11,30 @@
 #include <utility>
 #include <map>
 
-ME0PreRecoGaussianModel::ME0PreRecoGaussianModel(const edm::ParameterSet& config) :
-  ME0DigiPreRecoModel(config)
-, sigma_t(config.getParameter<double> ("timeResolution"))
-, sigma_u(config.getParameter<double> ("phiResolution"))
-, sigma_v(config.getParameter<double> ("etaResolution"))
-, corr(config.getParameter<bool> ("useCorrelation"))
-, etaproj(config.getParameter<bool> ("useEtaProjectiveGEO"))
-, digitizeOnlyMuons_(config.getParameter<bool> ("digitizeOnlyMuons"))
-, averageEfficiency_(config.getParameter<double> ("averageEfficiency"))
-, doBkgNoise_(config.getParameter<bool> ("doBkgNoise"))
-, simulateIntrinsicNoise_(config.getParameter<bool> ("simulateIntrinsicNoise"))
-, simulateElectronBkg_(config.getParameter<bool>("simulateElectronBkg"))
-, averageNoiseRate_(config.getParameter<double> ("averageNoiseRate"))
-, bxwidth_(config.getParameter<int> ("bxwidth"))
-, minBunch_(config.getParameter<int> ("minBunch"))
-, maxBunch_(config.getParameter<int> ("maxBunch"))
+const double cspeed = 29.9792458; // [cm/ns]
+const int bxwidth   = 25;         // [ns]
 
+ME0PreRecoGaussianModel::ME0PreRecoGaussianModel(const edm::ParameterSet& config) :
+  ME0DigiPreRecoModel(config), 
+  sigma_t(config.getParameter<double>("timeResolution")), 
+  sigma_u(config.getParameter<double>("phiResolution")), 
+  sigma_v(config.getParameter<double>("etaResolution")), 
+  corr(config.getParameter<bool>("useCorrelation")), 
+  etaproj(config.getParameter<bool>("useEtaProjectiveGEO")), 
+  digitizeOnlyMuons_(config.getParameter<bool>("digitizeOnlyMuons")), 
+  averageEfficiency_(config.getParameter<double>("averageEfficiency")), 
+  // simulateIntrinsicNoise_(config.getParameter<bool>("simulateIntrinsicNoise")),
+  // averageNoiseRate_(config.getParameter<double>("averageNoiseRate")), 
+  simulateElectronBkg_(config.getParameter<bool>("simulateElectronBkg")), 
+  simulateNeutralBkg_(config.getParameter<bool>("simulateNeutralBkg")), 
+  minBunch_(config.getParameter<int>("minBunch")), 
+  maxBunch_(config.getParameter<int>("maxBunch"))
 {
-  //params for the simple pol6 model of neutral and electron bkg for ME0:
-  ME0ModNeuBkgParam0 = 5.69e+06;
-  ME0ModNeuBkgParam1 = -293334;
-  ME0ModNeuBkgParam2 = 6279.6;
-  ME0ModNeuBkgParam3 = -71.2928;
-  ME0ModNeuBkgParam4 = 0.452244;
-  ME0ModNeuBkgParam5 = -0.0015191;
-  ME0ModNeuBkgParam6 = 2.1106e-06;
-
-  ME0ModElecBkgParam0 = 3.77712e+06;
-  ME0ModElecBkgParam1 = -199280;
-  ME0ModElecBkgParam2 = 4340.69;
-  ME0ModElecBkgParam3 = -49.922;
-  ME0ModElecBkgParam4 = 0.319699;
-  ME0ModElecBkgParam5 = -0.00108113;
-  ME0ModElecBkgParam6 = 1.50889e-06;
-
+  // polynomial parametrisation of neutral (n+g) and electron background
+  neuBkg.push_back(5.69e+06);     neuBkg.push_back(-293334);     neuBkg.push_back(6279.6); 
```

Comment:
do these parameters correspond to some universal law or cast in stone for some other reason?
If not, it seems more appropriate to define them via configuration parameters.